mage/query-modules/python/node-similarity.md

```diff
@@ -0,0 +1,210 @@
+
+**The Jaccard similarity** is computed using the following formula:
+<a href="https://www.codecogs.com/eqnedit.php?latex=Jaccard(A,B)&space;=&space;\frac{|A&space;\cap&space;B|}&space;{|A|&space;&plus;&space;|B|&space;-&space;|A&space;\cap&space;B|}" target="_blank"><img src="https://latex.codecogs.com/gif.latex?Jaccard(A,B)&space;=&space;\frac{|A&space;\cap&space;B|}&space;{|A|&space;&plus;&space;|B|&space;-&space;|A&space;\cap&space;B|}" title="Jaccard(A,B) = \frac{|A \cap B|} {|A| + |B| - |A \cap B|}" /></a>
+
+**The overalap similarity** is computed using the following formula:
+<a href="https://www.codecogs.com/eqnedit.php?latex=Overlap(A,B)&space;=&space;\frac{|A&space;\cap&space;B|}&space;{min(|A|,&space;|B|)}" target="_blank"><img src="https://latex.codecogs.com/gif.latex?Overlap(A,B)&space;=&space;\frac{|A&space;\cap&space;B|}&space;{min(|A|,&space;|B|)}" title="Overlap(A,B) = \frac{|A \cap B|} {min(|A|, |B|)}" /></a>
+
+Set A represents all neighbors of one node, set B represents all neighbors of the other node. In all the given formulas, the numerator is the cardinality of the intersection of set A and set B. The denominator differs but requires the cardinality of sets A and B in some way.
+
+The algorithm is implemented so that it ignores whether the graph is directed or undirected and treats the edges as if they were undirected. It also ignores multiple edges between two nodes and treats them as if there were only one edge.
+
+
+| Trait               | Value                                                 |
+| ------------------- | ----------------------------------------------------- |
+| **Module type**     | <Highlight color="#FB6E00">**algorithm**</Highlight>  |
+| **Implementation**  | <Highlight color="#FB6E00">**Python**</Highlight>     |
+| **Graph direction** | <Highlight color="#FB6E00">**undirected**</Highlight> |
+| **Edge weights**    | <Highlight color="#FB6E00">**unweighted**</Highlight> |
+| **Parallelism**     | <Highlight color="#FB6E00">**sequential**</Highlight> |
+
+## Procedures
+
+### `cosine(node1, node2, mode)`
+
+#### Input:
+
+* `node1: Union[Vertex, Tuple[Vertex]` ➡ The first node or a tuple of nodes.
+* `node2: Union[Vertex, Tuple[Vertex]]` ➡ The second node or a tuple of nodes.
+* `mode: str("cartesian")` ➡ If the given arguments are tuples, this argument determines whether to calculate the similarity between nodes pairwise ("pairwise" or "p") or calculate the similarity between one and each node ("cartesian" or "c"). The default value is "cartesian".
+
+#### Output:
+
+* `node1` ➡ The first node
+* `node2` ➡ The second node
+* `similarity` ➡  The cosine similarity between the first and the second node
+
```

Comment:
There should be usage section after each procedure

Reply:
Done.